docs/decisions/0004-axios-caching-implementation.rst

```diff
@@ -0,0 +1,87 @@
+
+   - For example changes made by a background job or an admin making changes
+
+ * The data being returned isn't specific to the authenticated user
+
+ * The data returned is specific to the authenticated user but front end does not make the request when there isn't an authenticated user
+
+It should not be used as the only caching strategy that gets used. It is preferable to use a serverside caching strategy in conjunction with frontend caching.
+
+How To Use Frontend Caching
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The frontend caching layer can be used by passing {useCache: true} into the function that gets the httpClient.
+
+e.g.
+    service.getAuthenticatedHttpClient({ useCache: true });
+
+By default the response for a GET request will be stored in IndexedDB for 5 minutes and that cached value will get invalidated on any POST, PUT, PATCH, or DELETE request made to the same url. The caching layer also works with the standard cache-control headers: max-age, no-cache, and no-store
+
+Each request can also override the default cache configurations. `The How To document about caching <https://github.com/edx/frontend-platform/blob/master/docs/how_tos/caching.rst>`_ has more detailed examples about how this caching layer can be used.
+
+
+Implementation Details
+~~~~~~~~~~~~~~~~~~~~~~
+
+For both versions of the current AxiosJwtAuthService HTTP clients, the authenticated client and the base unauthenticated clients, there are now two additional versions that use the cache (i.e. CachedAuthenticated and CachedUnauthenticated). Bringing the total number of httpClient instances up to four.
+In order to keep the interface more simple the cached clients don’t have their own getter functions exposing this detail, in order to use the cached http client micro frontends pass in a param for using the cache with the current getter functions.
+
+In the case of the cached clients failing to get created the uncached clients will be returned and used instead.
+
+The async hydrateAuthenticatedUser function will continue only ever using the uncached authenticated HTTP client so there won't be any concerns about the cache changing expected behavior of authentication.
+
+Only the AxiosJwtAuthService currently has cached http clients.
+
+Decisions
+---------
+
+By using the existing axios-cache-adapter we were able to avoid needing to implement our own caching strategy and forgo having micro frontends need to implement their own front end caching strategies.
+
+The default max-age for cached data was decided to be 5 minutes for now. The decision was based on it being a decent amount of the storage of cached data even for authentication specific requests. The assumption is made that sers won’t be frequently switching to different accounts on their browser frequently enough to have a wide impact and if they do then they will only see cached data for a maximum of 5 minutes. Micro frontends are also able to override the cache time to suit their own needs.
```

Comment:
Couple other points here:
* "The decision was based on it being a decent amount of the storage of cached data even for authentication specific requests." I'm not sure I understand this sentence, could you elaborate a little bit?
* You don't need the first "frequently"
* I think the third sentence could use a comma or two.
* "...override the cache time..." -> "...override the default cache expiry..."

Reply:
For the first bulletpoint the concern with authentication specific requests is data that's tied to the authenticated user still being used after the user logs out. Which if it happens that a user is still seeing their data after logging out it would only be for about 5 minutes.

I'll try figuring out a better way to word that